lib/puppet/provider/test_something/test_something.rb

```diff
@@ -0,0 +1,38 @@
+require 'puppet/resource_api/simple_provider'
+
+# Implementation for the test_something type using the Resource API.
+class Puppet::Provider::TestSomething::TestSomething < Puppet::ResourceApi::SimpleProvider
+  def get(_context)
+    # cause a hard error at runtime if the library is not installed yet
+    require 'cisco_node_utils'
+
+    @network_snmp ||= Cisco::SnmpServer.new
+
```

Comment:
Just more a question, can the require not just be required at the top, and then `@network_snmp ||= Cisco::SnmpServer.new` be put in an `initialize` method?

Reply:
I put it there because the provider needs to load, even if the node utils gem is not installed, which might happen in advanced scenarios. It might be interesting to see what *actually* happens when the provider loading fails - especially when `puppet device` is trying to configure a *different* device. We might want to test that once we have IOS and nexus on the go.